src/styles/converter.rs

```diff
@@ -0,0 +1,378 @@
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Class,
+                            value: node.as_class_selector().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_combinator() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Combinator,
+                            value: node.as_combinator().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_id_selector() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Id,
+                            value: node.as_id_selector().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_universal_selector() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Universal,
+                            value: "*".to_string(),
+                            ..Default::default()
+                        });
+                    } else if node.is_pseudo_class_selector() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::PseudoClass,
+                            value: node.as_pseudo_class_selector().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_pseudo_element_selector() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::PseudoElement,
+                            value: node.as_pseudo_element_selector().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_type_selector() {
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Type,
+                            value: node.as_type_selector().clone(),
+                            ..Default::default()
+                        });
+                    } else if node.is_attribute_selector() {
+                        let attr_selector = node.as_attribute_selector();
+                        selector.parts.push(CssSelectorPart {
+                            type_: CssSelectorType::Attribute,
+                            name: attr_selector.0.clone(),
+                            matcher: MatcherType::Equals, // @todo: this needs to be parsed
+                            value: attr_selector.2.clone(),
+                            flags: attr_selector.3.clone(),
+                        });
+                    } else {
+                        panic!("Unknown selector type: {:?}", node);
+                    }
```

Comment:
Why not use a match statement here?

```rust
let part = match &*node.node_type {
    NodeType::Ident { value} => CssSelectorPart {
        type_: CssSelectorType::Type,
        value: value.clone(),
        ..Default::default()
    },
    _ => {
        todo!("Implement the rest of the node types");
    }
};

selector.parts.push(part);
                    
```

Reply:
how would one do this, when there are multipe types available?

Reply:
What do you mean?

Reply:
ok.. fixed